### DISCUSSION TITLE
Camera Settings Control for K1 Max with new hardware version

### DISCUSSION THREAD

**Discussion by loosegoosage** [Guilouz/Creality-Helper-Script-Wiki#444] [Features Request]
Curious if this is something that can be requested or not.  I am going to assume it is not a bug, but rather a case of being unsupported, since the script basically states that "You have the new hardware version of the camera and it's not compatible!"

**Comment by Hammerfest** [comment]
That is correct!

Creality changed the camera module hardware, as such the new hardware does not respond to the commands exposed on the older hardware.

If anyone is a hardware/firmware hacker and can find out what commands it can/does accept or MAKE it accept commands it would be welcomed greatly.

IMHO, the new camera sucks by comparison, if you add extra lighting you cannot make adjustments to compensate :(

**Reply by loosegoosage** [reply]
Well thats dookie, guess they are trying to cut costs or something.  I will have to live without for now unless i can break down its barrier..... :(

**Comment by mfraser** [comment]
Does lsusb reveal any information on the new camera? The older one is this http://linux-hardware.org/?id=usb:0c45-6366

**Reply by loosegoosage** [reply]
Not much **lsusb** output,
_Bus 001 Device 003: ID a108:2231
Bus 001 Device 005: ID 1a86:55d3 QinHeng Electronics USB2.0 Hub
Bus 001 Device 002: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub_

This is **v4l2-ctl --list-devices** output,
_jz-rot ():
        /dev/video0

CCX2F3299 (1.3):
        /dev/media0

Dummy video device (0x0000) (platform:v4l2loopback-000):
        /dev/video3

CCX2F3299 (usb-13500000.otg_new-1.3):
        /dev/video4

vpu-felix (vpu-felix):
        /dev/video2

vpu-helix (vpu-helix):
        /dev/video1_

Then this is output from **v4l2-ctl --device=/dev/video4 --all**,
_Driver Info:
        Driver name      : uvcvideo
        Card type        : CCX2F3299
        Bus info         : usb-13500000.otg_new-1.3
        Driver version   : 4.4.94
        Capabilities     : 0x84200001
                Video Capture
                Streaming
                Extended Pix Format
                Device Capabilities
        Device Caps      : 0x04200001
                Video Capture
                Streaming
                Extended Pix Format
Media Driver Info:
        Driver name      : uvcvideo
        Model            : CCX2F3299
        Serial           : 1234567890
        Bus info         : 1.3
        Media version    : 0.1.0
        Hardware revision: 0x00000310 (784)
        Driver version   : 4.4.94
Entity Info:
        ID               : 1
        Name             : CCX2F3299
        Type             : V4L2 I/O
        Flags            : default
        Major            : 81
        Minor            : 4
        Pad              : 0: Sink
Priority: 2
Video input : 0 (Camera 1: ok)
Format Video Capture:
        Width/Height      : 1280/720
        Pixel Format      : 'MJPG' (Motion-JPEG)
        Field             : None
        Bytes per Line    : 0
        Size Image        : 1572864
        Colorspace        : Default
        Transfer Function : Default (maps to Rec. 709)
        YCbCr/HSV Encoding: Default (maps to ITU-R 601)
        Quantization      : Default (maps to Full Range)
        Flags             :
Crop Capability Video Capture:
        Bounds      : Left 0, Top 0, Width 1280, Height 720
        Default     : Left 0, Top 0, Width 1280, Height 720
        Pixel Aspect: 1/1
Selection Video Capture: crop_default, Left 0, Top 0, Width 1280, Height 720, Flags:
Selection Video Capture: crop_bounds, Left 0, Top 0, Width 1280, Height 720, Flags:
Streaming Parameters Video Capture:
        Capabilities     : timeperframe
        Frames per second: 15.000 (15/1)
        Read buffers     : 0_

**Reply by loosegoosage** [reply]
Definitely appears different.

**Reply by mfraser** [reply]
This is from the older K1Max.

lsusb
```
Bus 001 Device 003: ID 0c45:6366 Microdia 
Bus 001 Device 005: ID 1a86:55d3 QinHeng Electronics USB2.0 Hub
Bus 001 Device 002: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

v4l2-ctl --list-devices 
```
jz-rot ():
        /dev/video0

CREALITY CAM (1.3):
        /dev/media0

Dummy video device (0x0000) (platform:v4l2loopback-000):
        /dev/video3

CREALITY CAM (usb-13500000.otg_new-1.3):
        /dev/video4
        /dev/video5

vpu-felix (vpu-felix):
        /dev/video2

vpu-helix (vpu-helix):
        /dev/video1
```
v4l2-ctl --device=/dev/video4 --all
```
Driver Info:
        Driver name      : uvcvideo
        Card type        : CREALITY CAM
        Bus info         : usb-13500000.otg_new-1.3
        Driver version   : 4.4.94
        Capabilities     : 0x84200001
                Video Capture
                Streaming
                Extended Pix Format
                Device Capabilities
        Device Caps      : 0x04200001
                Video Capture
                Streaming
                Extended Pix Format
Media Driver Info:
        Driver name      : uvcvideo
        Model            : CREALITY CAM
        Serial           : SN0001
        Bus info         : 1.3
        Media version    : 0.1.0
        Hardware revision: 0x00000100 (256)
        Driver version   : 4.4.94
Entity Info:
        ID               : 1
        Name             : CREALITY CAM
        Type             : V4L2 I/O
        Flags            : default
        Major            : 81
        Minor            : 4
        Pad              : 0: Sink
Priority: 2
Video input : 0 (Camera 1: ok)
Format Video Capture:
        Width/Height      : 1280/720
        Pixel Format      : 'MJPG' (Motion-JPEG)
        Field             : None
        Bytes per Line    : 0
        Size Image        : 1843789
        Colorspace        : Default
        Transfer Function : Default (maps to Rec. 709)
        YCbCr/HSV Encoding: Default (maps to ITU-R 601)
        Quantization      : Default (maps to Full Range)
        Flags             : 
Crop Capability Video Capture:
        Bounds      : Left 0, Top 0, Width 1280, Height 720
        Default     : Left 0, Top 0, Width 1280, Height 720
        Pixel Aspect: 1/1
Selection Video Capture: crop_default, Left 0, Top 0, Width 1280, Height 720, Flags: 
Selection Video Capture: crop_bounds, Left 0, Top 0, Width 1280, Height 720, Flags: 
Streaming Parameters Video Capture:
        Capabilities     : timeperframe
        Frames per second: 15.000 (15/1)
        Read buffers     : 0
                     brightness 0x00980900 (int)    : min=-64 max=64 step=1 default=-8193 value=0
                       contrast 0x00980901 (int)    : min=0 max=64 step=1 default=57343 value=32
                     saturation 0x00980902 (int)    : min=0 max=128 step=1 default=57343 value=56
                            hue 0x00980903 (int)    : min=-40 max=40 step=1 default=-8193 value=0
 white_balance_temperature_auto 0x0098090c (bool)   : default=1 value=1
                          gamma 0x00980910 (int)    : min=72 max=500 step=1 default=57343 value=80
                           gain 0x00980913 (int)    : min=0 max=100 step=1 default=57343 value=0
           power_line_frequency 0x00980918 (menu)   : min=0 max=2 default=1 value=1
                                0: Disabled
                                1: 50 Hz
                                2: 60 Hz
      white_balance_temperature 0x0098091a (int)    : min=2800 max=6500 step=1 default=57343 value=4600 flags=inactive
                      sharpness 0x0098091b (int)    : min=0 max=6 step=1 default=57343 value=3
         backlight_compensation 0x0098091c (int)    : min=0 max=2 step=1 default=57343 value=1
                  exposure_auto 0x009a0901 (menu)   : min=0 max=3 default=0 value=3
                                1: Manual Mode
                                3: Aperture Priority Mode
              exposure_absolute 0x009a0902 (int)    : min=1 max=5000 step=1 default=156 value=156 flags=inactive
         exposure_auto_priority 0x009a0903 (bool)   : default=0 value=0

```
Interestingly, v4l2-ctl --device=/dev/video5 --all shows that can output H.264 too:
```
Driver Info:
        Driver name      : uvcvideo
        Card type        : CREALITY CAM
        Bus info         : usb-13500000.otg_new-1.3
        Driver version   : 4.4.94
        Capabilities     : 0x84200001
                Video Capture
                Streaming
                Extended Pix Format
                Device Capabilities
        Device Caps      : 0x04200001
                Video Capture
                Streaming
                Extended Pix Format
Media Driver Info:
        Driver name      : uvcvideo
        Model            : CREALITY CAM
        Serial           : SN0001
        Bus info         : 1.3
        Media version    : 0.1.0
        Hardware revision: 0x00000100 (256)
        Driver version   : 4.4.94
Entity Info:
        ID               : 2
        Name             : CREALITY CAM
        Type             : V4L2 I/O
        Pad              : 0: Sink
Priority: 2
Video input : 0 (Camera 1: ok)
Format Video Capture:
        Width/Height      : 1920/1080
        Pixel Format      : 'H264' (H.264)
        Field             : None
        Bytes per Line    : 3840
        Size Image        : 0
        Colorspace        : sRGB
        Transfer Function : Default (maps to sRGB)
        YCbCr/HSV Encoding: Default (maps to ITU-R 601)
        Quantization      : Default (maps to Full Range)
        Flags             : 
Crop Capability Video Capture:
        Bounds      : Left 0, Top 0, Width 1920, Height 1080
        Default     : Left 0, Top 0, Width 1920, Height 1080
        Pixel Aspect: 1/1
Selection Video Capture: crop_default, Left 0, Top 0, Width 1920, Height 1080, Flags: 
Selection Video Capture: crop_bounds, Left 0, Top 0, Width 1920, Height 1080, Flags: 
Streaming Parameters Video Capture:
        Capabilities     : timeperframe
        Frames per second: invalid (1/0)
        Read buffers     : 0
                     brightness 0x00980900 (int)    : min=-64 max=64 step=1 default=-8193 value=0
                       contrast 0x00980901 (int)    : min=0 max=64 step=1 default=57343 value=32
                     saturation 0x00980902 (int)    : min=0 max=128 step=1 default=57343 value=56
                            hue 0x00980903 (int)    : min=-40 max=40 step=1 default=-8193 value=0
 white_balance_temperature_auto 0x0098090c (bool)   : default=1 value=1
                          gamma 0x00980910 (int)    : min=72 max=500 step=1 default=57343 value=80
                           gain 0x00980913 (int)    : min=0 max=100 step=1 default=57343 value=0
           power_line_frequency 0x00980918 (menu)   : min=0 max=2 default=1 value=1
                                0: Disabled
                                1: 50 Hz
                                2: 60 Hz
      white_balance_temperature 0x0098091a (int)    : min=2800 max=6500 step=1 default=57343 value=4600 flags=inactive
                      sharpness 0x0098091b (int)    : min=0 max=6 step=1 default=57343 value=3
         backlight_compensation 0x0098091c (int)    : min=0 max=2 step=1 default=57343 value=1
                  exposure_auto 0x009a0901 (menu)   : min=0 max=3 default=0 value=3
                                1: Manual Mode
                                3: Aperture Priority Mode
              exposure_absolute 0x009a0902 (int)    : min=1 max=5000 step=1 default=156 value=156 flags=inactive
         exposure_auto_priority 0x009a0903 (bool)   : default=0 value=0
```

**Reply by mfraser** [reply]
Looks like is the same camera as the Creality Nebula 20 https://community.octoprint.org/t/usb-webcams-known-to-work-with-mjpg-streamer/21149/19:

Camera 	USBID 	camera_usb_options 	Tester Comments
[Creality Nebula 20](https://store.creality.com/products/creality-nebula-camera) 	a108:2231 	-r 1920x1080 	auto IR night

**Reply by loosegoosage** [reply]
Does that printers camera work in klipper?

**Reply by spitfire** [reply]
If I understand it correctly that is the same as the one dedicated for Creality Ender 3 v3 KE, which is based on Klipper, and pretty similar to K1

**Reply by loosegoosage** [reply]
Ok, i will see what i can figure out maybe somehow its doable.

**Reply by loosegoosage** [reply]
@mfraser it looks to be the exact same driver, im guessing it has dual support for 720 and 1080 on the old? odd it shows as 2 devices with slightly different specs.

**Reply by spitfire** [reply]
on Creality Ender 3 v3 KE with the Nebula camera it is showing:

```
[#] lsusb
Bus 001 Device 006: ID a108:2231  USB2.0 Hub
Bus 001 Device 004: ID 05e3:0610 Genesys Logic, Inc. 4-port hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

and 

```
[#] v4l2-ctl --device=/dev/video4 --all
Driver Info:
	Driver name      : uvcvideo
	Card type        : CCX2F3298
	Bus info         : usb-13500000.otg_new-1.1
	Driver version   : 4.4.94
	Capabilities     : 0x84200001
		Video Capture
		Streaming
		Extended Pix Format
		Device Capabilities
	Device Caps      : 0x04200001
		Video Capture
		Streaming
		Extended Pix Format
Media Driver Info:
	Driver name      : uvcvideo
	Model            : CCX2F3298
	Serial           : 1234567890
	Bus info         : 1.1
	Media version    : 0.1.0
	Hardware revision: 0x00000310 (784)
	Driver version   : 4.4.94
Entity Info:
	ID               : 1
	Name             : CCX2F3298
	Type             : V4L2 I/O
	Flags            : default
	Major            : 81
	Minor            : 4
	Pad              : 0: Sink
Priority: 2
Video input : 0 (Camera 1: ok)
Format Video Capture:
	Width/Height      : 1920/1080
	Pixel Format      : 'MJPG' (Motion-JPEG)
	Field             : None
	Bytes per Line    : 0
	Size Image        : 1572864
	Colorspace        : Default
	Transfer Function : Default (maps to Rec. 709)
	YCbCr/HSV Encoding: Default (maps to ITU-R 601)
	Quantization      : Default (maps to Full Range)
	Flags             :
Crop Capability Video Capture:
	Bounds      : Left 0, Top 0, Width 1920, Height 1080
	Default     : Left 0, Top 0, Width 1920, Height 1080
	Pixel Aspect: 1/1
Selection Video Capture: crop_default, Left 0, Top 0, Width 1920, Height 1080, Flags:
Selection Video Capture: crop_bounds, Left 0, Top 0, Width 1920, Height 1080, Flags:
Streaming Parameters Video Capture:
	Capabilities     : timeperframe
	Frames per second: 15.000 (15/1)
	Read buffers     : 0
                     brightness 0x00980900 (int)    : min=50 max=160 step=1 default=-8193 value=128
                       contrast 0x00980901 (int)    : min=50 max=160 step=1 default=57343 value=128
                     saturation 0x00980902 (int)    : min=50 max=160 step=1 default=57343 value=128
                            hue 0x00980903 (int)    : min=50 max=160 step=1 default=-8193 value=128
 white_balance_temperature_auto 0x0098090c (bool)   : default=1 value=1
```

which looks very similar to the first one, no video5 or H.264 anywhere though.

**Comment by spitfire** [comment]
Is the new revision of camera better in any way?

**Reply by loosegoosage** [reply]
Tbh, I never seen the old nor did I look up to compare.  I would hope so, but then again like I said in an earlier reply I wouldn't doubt them just cutting cost.

**Comment by rocklee5** [comment]
I have the old Camera and I am facing this error " You have the new hardware version of the camera and it's not compatible!" when trying to install the camera control script.

**Comment by MzTechnology97** [comment]
I just tried to connect to the camera via serial, I soldered some wires to the PCB on TX, RX and GND and I managed to connect. Unfortunately access is password protected.

Serial init OK!jz_serial_init ok
pll_init:366
pll_cfg.pdiv = 10, pll_cfg.h2div = 5, pll_cfg.h0div = 5, pll_cfg.cdiv = 1, pll_c                                                                                                                   fg.l2div = 2
nf=84 nr = 1 od0 = 1 od1 = 2
cppcr is 05405100
CPM_CPAPCR 0540510d
nf=84 nr = 1 od0 = 1 od1 = 2
cppcr is 05405100
CPM_CPMPCR 07d0590d
nf=110 nr = 1 od0 = 1 od1 = 2
cppcr is 06e05100
CPM_CPVPCR 06e0510d
cppcr 0x9a773310
apll_freq 1008000000
mpll_freq 1000000000
vpll_freq = 1320000000
ddr sel mpll, cpu sel apll
ddrfreq 500000000
cclk  1008000000
l2clk 504000000
h0clk 200000000
h2clk 200000000
pclk  100000000
sdram init start
REG_DDR_LMR: 00000210
REG_DDR_LMR: 00000310
REG_DDR_LMR: 00000110
REG_DDR_LMR, MR0: 00f73011
T31_0x5: 00000007
T31_0x15: 0000000c
T31_0x4: 00000000
T31_0x14: 00000002
INNO_TRAINING_CTRL 1: 00000000
INNO_TRAINING_CTRL 2: 000000a1
T31_cc: 00000003
INNO_TRAINING_CTRL 3: 000000a0
T31_118: 0000003c
T31_158: 0000003c
T31_190: 00000020
T31_194: 0000001f
DDR PHY init OK
DDR Controller init
SDRAM init ok
#######norflash id 00204017#####
#######norflash model XM25QH64C#####

spl_sfc_nor_load_kernel ...
boot uImage...

uImage size = [3628865]
OK
images->ep = 802699c0
[    0.000000] Initializing cgroup subsys cpu
[    0.000000] Initializing cgroup subsys cpuacct
[    0.000000] Linux version 3.10.14__isvp_swan_1.0__ (root@ubuntu) (gcc version                                                                                                                    4.7.2 (Ingenic r2.3.3 2016.12) ) #1 PREEMPT Tue Aug 22 09:55:24 CST 2023
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 RESET ERROR PC:D0004154
[    0.000000] CPU0 revision is: 00d00100 (Ingenic Xburst)
[    0.000000] FPU revision is: 00b70000
[    0.000000] CCLK:1008MHz L2CLK:504Mhz H0CLK:250MHz H2CLK:250Mhz PCLK:125Mhz
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 0033b000 @ 00010000 (usable)
[    0.000000]  memory: 00465000 @ 0034b000 (usable after init)
mdev is ok......
ifconfig: socket: Address family not supported by protocol
/system/init/app_init.sh: line 8: anticopy: not found

Ingenic-uc1_1 login: INFO(KIVA): AntiCopy_Verify ...
INFO: anticopy verify OK!
INFO(KIVA): AntiCopy_Verify ...ok
INFO(KIVA): kiva_load_config_init ...
INFO(KIVA): kiva_load_config_init ...ok
INFO(KIVA): kiva version = 2.0.0
INFO(KIVA): ucamera_system_init ...
INFO(KIVA): ucamera_system_init ...ok
INFO(KIVA): module_led_init ...
INFO(KIVA): module_led_init ...ok
open config file/system/config/update.config failed
can't load update.config
INFO(KIVA): module_ucamera_init ...
[Ucamera]: [INFO]: Video Start.

INFO(KIVA): module_ucamera_init ...ok
INFO(KIVA): module_autofocus_init ...
INFO(KIVA): module_autofocus_init ...ok
INFO(KIVA): kiva_process_init ...
INFO(KIVA): kiva_process_init ...ok
---- FPGA board is ready ----
  Board UID : 30AB6E51
  Board HW ID : 72000460
  Board rev.  : 5DE5A975
  Board date  : 20190326
-----------------------------
[Ucamera]: UVC: Setting format to: MJPG 1920x1080

[Ucamera]: UVC: uvc event stream on!

ERROR(UVC_CONTROL): imp sys not ready, wait and retry:1
INFO(UVC_CONTROL): set Antiflicker level:1 fps:25
INFO(UVC_CONTROL): set Fixedcontrastattr mode = 0 OK!
[Ucamera]: UVC: Starting video stream.


Ingenic-uc1_1 login: root
Password:
Login incorrect
![photo1711902928](https://github.com/Guilouz/Creality-K1-Series/assets/86824210/23dce148-6705-4b67-8052-e00c904ed0df)

![photo1711902816](https://github.com/Guilouz/Creality-K1-Series/assets/86824210/cb3c5b61-c4ff-435a-8da6-e4e73d58fc52)

**Reply by Hammerfest** [reply]
A quick search shows at least 3 instances where there is a hardcoded password, maybe creality or ingenic didnt change it?

"ismart12"

**Reply by MzTechnology97** [reply]
I have already tried several generic passwords including this one, but unfortunately it is incorrect. You should do a dump of the chip and try to find out the password or change it.
If I can, I can try to make a dump and share it. Maybe someone can find it.

**Reply by Hammerfest** [reply]
would the firmware be on that XMC(?) chip next to the T31 or would it be on the T31 itself?

**Reply by MzTechnology97** [reply]
on the chip next to the t31. XM25QH64C chip flash 8 MB

**Comment by spitfire** [comment]
![image](https://github.com/Guilouz/Creality-K1-Series/assets/42872/7a51309a-33bc-42fd-b2bf-c0b1180c5f77)
Is this new or old? I guess it’s new some it says it’s unsupported

**Reply by MzTechnology97** [reply]
it looks the same as mine. So it should be the new version.

**Comment by MzTechnology97** [comment]
I just did a rom dump.
I share it, maybe someone can do something.
https://drive.google.com/file/d/1BTP2MI3MZi9E_O_2lNqhfJjOpJXjcjWm/view?usp=sharing

**Reply by MzTechnology97** [reply]
Here's what I managed to achieve with binwalk. I can't figure out where the U-boot partition is located.
I assume it is compressed and the address could be 0x4DE4

DECIMAL     HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
17572         0x44A4          CRC32 polynomial table, little endian
19940         0x4DE4          LZO compressed data
32768         0x8000          uImage header, header size: 64 bytes, header CRC: 0xBE9D47AD, created: 2023-08-22 01:55:27, image size: 3628865 bytes, Data Address: 0x80010000, Entry Point: 0x802699C0, data CRC: 0x50F1894, OS: Linux, CPU: MIPS, image type: OS Kernel Image, image name: "Linux-3.10.14__isvp_swan_1.0__"
32832         0x8040          LZO compressed data
710387        0xAD6F3         Certificate in DER format (x509 v3), header length: 4, sequence length: 12621
2225163       0x21F40B        mcrypt 2.5 encrypted data, algorithm: ")l", keysize: 4076 bytes, mode: "2",
3317944       0x32A0B8        mcrypt 2.5 encrypted data, algorithm: "L", keysize: 20844 bytes, mode: "]",
3964928       0x3C8000        JFFS2 filesystem, little endian
3965724       0x3C831C        Zlib compressed data, compressed
3968044       0x3C8C2C        Zlib compressed data, compressed
3970432       0x3C9580        Zlib compressed data, compressed
3970760       0x3C96C8        JFFS2 filesystem, little endian

**Reply by MzTechnology97** [reply]
I managed to extract the partition and found the following credentials. I haven't tested whether they work yet. lhgx168:qdGt8T0XjiWXI:0:0:Linux User,,,:/home:/bin/sh

**Reply by Hammerfest** [reply]
qdGt8T0XjiWXI is normally empty password, thats wild

partially incorrect:

https://www.seancassidy.me/etc/passwords.txt

**Reply by MzTechnology97** [reply]
I logged in with username lhgx168 and blank password

**Reply by MzTechnology97** [reply]
![image](https://github.com/Guilouz/Creality-K1-Series/assets/86824210/1e09134c-50a4-430a-b67b-2036eb5f0fad)

**Reply by Hammerfest** [reply]
EDIT: I read that wrong, brain auto-inserted a "did not work" and when I saw it in the image, brain stopped go boom...

**Reply by MzTechnology97** [reply]
I found the file to change the camera parameters.

/system/config/uvc.attr

powerline_freq:0001
contrast:0128
brightness:0128
saturation:0128
sharpness:0128
hue:0128
white_balance:0168
white_balance_auto:0001
backlight:0000
gamma:0400

**Reply by spitfire** [reply]
Is there a way to enable WebRTC on it (I believe the older model had it)?

**Reply by MzTechnology97** [reply]
There will be a way. But I have no idea how to do that at the moment.

**Reply by spitfire** [reply]
I was asking, cause I have no idea how hardware-dependent (i.e. requiring hardware acceleration) that might be

**Reply by MzTechnology97** [reply]
the Ingenic t31 chip is actually powerful enough to do this. It is often used for IP surveillance cameras with many more functions than a banal USB webcam.

**Reply by Hammerfest** [reply]
sweet, hopefully we can probe all the goodness and get at least some of the auto adjust controls back if not full resolution/fps/etc

I am no linux user so I was watching https://youtu.be/FnhkprszfcU?t=490
thats a pretty impressive spec sheet for the paltry output the creality board is able to handle

**Reply by MzTechnology97** [reply]
In the end I decided to completely remove the creality camera and replaced it with a good webcam of much better quality and I can easily set the image parameters.

**Reply by spitfire** [reply]
I have no intention of doing that, and I hoped it could work with Creality's camera. I think I'd rather have what I do now, rather than buying another one and trying to safely mount it inside.

**Reply by Hammerfest** [reply]
totally understand not wanting to futz with it MzT, your route is the best route if ya can afford it

thank you for at least opening the door, hopefully someone more knowledgeable then I takes interest!
VERY MUCH APPRECIATED <3

**Reply by spitfire** [reply]
So what did you get/how did you mount it?

**Reply by MzTechnology97** [reply]
I used a webcam that I used with my old printer. I simply disassembled it and adapted it to be placed in almost the same position as the original and changed the lens to a 140 degree one.
https://a.co/d/0IyVSUK
